datalake/src/main/java/com/sequenceiq/datalake/flow/salt/rotatepassword/RotateSaltPasswordTrackerActions.java

```diff
@@ -0,0 +1,104 @@
+import java.util.Optional;
+
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.statemachine.action.Action;
+
+import com.sequenceiq.cloudbreak.common.event.Selectable;
+import com.sequenceiq.datalake.entity.DatalakeStatusEnum;
+import com.sequenceiq.datalake.flow.SdxContext;
+import com.sequenceiq.datalake.flow.SdxEvent;
+import com.sequenceiq.datalake.flow.salt.rotatepassword.event.RotateSaltPasswordFailureResponse;
+import com.sequenceiq.datalake.flow.salt.rotatepassword.event.RotateSaltPasswordWaitRequest;
+import com.sequenceiq.datalake.flow.salt.rotatepassword.event.RotateSaltPasswordSuccessResponse;
+import com.sequenceiq.datalake.service.AbstractSdxAction;
+import com.sequenceiq.datalake.service.sdx.status.SdxStatusService;
+
+@Configuration
+public class RotateSaltPasswordTrackerActions {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(RotateSaltPasswordTrackerActions.class);
+
+    @Inject
+    private SdxStatusService sdxStatusService;
+
+    @Bean(name = "ROTATE_SALT_PASSWORD_WAITING_STATE")
+    public Action<?, ?> rotateSaltPasswordTracking() {
+        return new AbstractSdxAction<>(SdxEvent.class) {
+
+            @Override
+            protected void doExecute(SdxContext context, SdxEvent payload, Map<Object, Object> variables) throws Exception {
+                LOGGER.info("Waiting for rotate salt password for SDX stack {}", context.getSdxId());
+                sdxStatusService.setStatusForDatalakeAndNotify(DatalakeStatusEnum.SALT_PASSWORD_ROTATION_IN_PROGRESS,
+                        "Rotating salt password", context.getSdxId());
+                sendEvent(context);
+            }
+
+            @Override
+            protected Object getFailurePayload(SdxEvent payload, Optional<SdxContext> flowContext, Exception ex) {
+                return new RotateSaltPasswordFailureResponse(payload.getResourceId(), payload.getUserId(), ex);
+            }
+
+            @Override
+            protected Selectable createRequest(SdxContext context) {
+                return new RotateSaltPasswordWaitRequest(context.getSdxId(), context.getUserId());
+            }
+        };
+    }
+
+    @Bean(name = "ROTATE_SALT_PASSWORD_SUCCESS_STATE")
+    public Action<?, ?> rotateSaltPasswordFinished() {
+        return new AbstractSdxAction<>(RotateSaltPasswordSuccessResponse.class) {
+
+            @Override
+            protected void doExecute(SdxContext context, RotateSaltPasswordSuccessResponse payload, Map<Object, Object> variables) throws Exception {
+                LOGGER.info("Rotating salt password for SDX stack {} finished", context.getSdxId());
+                sdxStatusService.setStatusForDatalakeAndNotify(DatalakeStatusEnum.RUNNING,
+                        "Rotated salt password successfully", context.getSdxId());
```

Comment:
Nitpick: Maybe "Successfully rotated the password for the 'salt' user." sounds a little less minimalist.

In general 'salt password' sounds like slang internal to CB devs and feels weird in customer facing messages. (Especially because the similar expression 'password salt' has a totally different meaning, but also related to password based security).

Reply:
Actually the username is 'saltuser', but it seems like an implementation detail to me. How about referring to it as SaltStack user?

Reply:
That's fine too.